orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,207 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
+
+	// if there are no errors, it means that the symlink is present, nothing to do
+	if err == nil {
+		return nil
+	}
+
+	// handling error by renaming the locked binary file, marking it for deletion on reboot and
+	// regenerating the symlink
+	if !errors.Is(err, os.ErrNotExist) {
+
+		// We are now about to perform a sensitive operation
+
+		// renaming locked binary to a different file, the process will keep running, but it will be renamed
+		// target orbit process is not terminated on purpose to avoid potential erros
+		temporaryOrbitPath := orbitPath + "." + strings.ToUpper(uuid.New().String())
+
+		if err := os.Rename(orbitPath, temporaryOrbitPath); err != nil {
+			return fmt.Errorf("rename: %w", err)
+		}
+
+		// we need the symlink check to pass, so we are regenerating it to the newly renamed orbit binary.
+		// We avoid using child directories here to reduce logic complexity.
+		// The symlink is going to be regenerated and deleted during update process
+		if err := os.Symlink(temporaryOrbitPath, orbitPath); err != nil {
+			return fmt.Errorf("symlink current: %w", err)
+		}
+
+		// the renamed binary file is locked because is used by a running process
+		// so only thing possible is to mark it to be deleted upon reboot by using MOVEFILE_DELAY_UNTIL_REBOOT flag
+		// https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexw
+		if err := windows.MoveFileEx(windows.StringToUTF16Ptr(temporaryOrbitPath), nil, windows.MOVEFILE_DELAY_UNTIL_REBOOT); err != nil {
+			return fmt.Errorf("movefileex: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// isRunningAsSystem checks if the current process is running as SYSTEM
+func isRunningAsSystem() (bool, error) {
+	// getting the current process token
+	token, err := windows.OpenCurrentProcessToken()
+	if err != nil {
+		return false, err
+	}
+	defer token.Close()
+
+	// getting the current process user
+	user, err := token.GetTokenUser()
+	if err != nil {
+		return false, err
+	}
+
+	// checking if the current process user is SYSTEM
+	if windows.EqualSid(user.User.Sid, constant.SystemSID) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// isRunningFromStagingDir checks if the current process is running from the staging directory
+func isRunningFromStagingDir() (bool, error) {
+	// getting current working directory
+	execPath, err := getExecutablePath()
+	if err != nil {
+		return false, err
+	}
+
+	// checking if the current executable directory is the staging directory and return error otherwise
+	if !strings.HasSuffix(strings.ToLower(execPath), "staging") {
+		return false, errors.New("not running from the staging directory")
+	}
+
+	return true, nil
+}
+
+// shouldQuirksRun determines if the software update quirks should be run
+// by checking if process is running as system and from staging directory
+// we can relax the constrains a bit if needed and just check for SYSTEM execution context
+func shouldQuirksRun() bool {
+	isSystem, err := isRunningAsSystem()
```

Comment:
Q: Does Orbit ever run as non-system on production? (or during development/testing)

Reply:
yes, customers that run `orbit shell` knows that orbit can be run from the command line, so they might end up running `orbit --help` from non-system context.

Reply:
Oh good point.